### ISSUE TITLE
Error saving notification record! - java.lang.NullPointerException

### ISSUE THREAD

**Issue by tapz** [OneSignal/OneSignal-Android-SDK#264]
When the app is active and it receives a notification the app crashes. Version 3.5.8.

```
07-13 22:39:35.277 30702-30702/? E/OneSignal: Error saving notification record! 
java.lang.NullPointerException: 
Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' 
on a null object reference
  at com.onesignal.NotificationBundleProcessor.saveNotification(NotificationBundleProcessor.java:164)
  at com.onesignal.NotificationBundleProcessor.saveNotification(NotificationBundleProcessor.java:112)
  at com.onesignal.NotificationBundleProcessor.processBundle(NotificationBundleProcessor.java:400)
  at com.onesignal.GcmBroadcastReceiver.processOrderBroadcast(GcmBroadcastReceiver.java:95)
  at com.onesignal.GcmBroadcastReceiver.onReceive(GcmBroadcastReceiver.java:61)
  at android.app.ActivityThread.handleReceiver(ActivityThread.java:3032)
  at android.app.ActivityThread.-wrap18(ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:154)
  at android.app.ActivityThread.main(ActivityThread.java:6123)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
```

**Comment by jkasten2** [comment]
@tapz Thanks for reporting this. Could you also include the payload of the notification, if create from the OneSignal REST API or the exact setting you used on the dashboard.

Please include any OneSignal initialization customizations you may have as well in your app's code.

**Comment by twanschik** [comment]
Same problem here:
Error saving notification record! 
                                                                             java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' on a null object reference
                                                                                 at com.onesignal.NotificationBundleProcessor.saveNotification(NotificationBundleProcessor.java:166)
                                                                                 at com.onesignal.NotificationExtenderService.processJsonObject(NotificationExtenderService.java:181)
                                                                                 at com.onesignal.NotificationExtenderService.processIntent(NotificationExtenderService.java:142)
                                                                                 at com.onesignal.NotificationExtenderService.onHandleIntent(NotificationExtenderService.java:110)
                                                                                 at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:68)
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                                 at android.os.Looper.loop(Looper.java:164)
                                                                                 at android.os.HandlerThread.run(HandlerThread.java:65)

Server side:
data = json.dumps({
	'app_id': customer.os_app_id,
	'included_segments': ['All'],
	'content_available': True,
})
connection.request('POST', '/api/v1/notifications', data, {
	'Authorization': 'Basic ' + customer.os_rest_api_key,
	'Content-Type': 'application/json; charset=utf-8',
})

Seems like 'content_available': True isn't handled correctly by the latest SDK ('com.onesignal:OneSignal:[3.6.0, 3.99.99]'). When setting 'contents' the app doesn't crash.

**Comment by tapz** [comment]
The app either crashes or displays the _silent_ notification. Both of these are totally unacceptable.

**Comment by jkasten2** [comment]
@tapz @twanschik Fixed this in version 3.6.2.
